### PR TITLE
feat: add notifications to the demo page

### PR DESCRIPTION
**Pull request by drodil**
Adds notifications to the demo page (even though there are no plugins sending them yet but just to give idea what it looks like)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ The following plugins have been added to help better illustrate Backstage featur
 - [Home](https://demo.backstage.io/home)
 - [Tech Radar](https://demo.backstage.io/tech-radar)
 - [To Do](https://demo.backstage.io/catalog/default/component/backstage-demo/todos)
+- [Notifications](https://demo.backstage.io/notifications)
```
Comment (awanlin) [inline README.md:37]: Can you create an Alpha subheading in this section? As Notifications is pretty new I think that makes more sense to me. Can we also perhaps include a link to the related BEP?

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this @drodil, this makes sense to me. My only concern is that Notifications are still in an alpha stage so I think we should at least mention that.

**Review by awanlin — CHANGES_REQUESTED** [review]
Hi @drodil, can you rebase to resolve the conflicts? Then I can approve and merge this in 👍

**Comment by awanlin** [comment]
Hi @drodil, any chance you can rebase to resolve the conflict? Then we can merge it in. 🚀

**Comment by drodil** [comment]
@awanlin rebased!

**Review by awanlin — APPROVED** [review]

**Pull request merged**